src/search/engines/startpage/startpage.go

```diff
@@ -26,28 +25,15 @@ func Search(ctx context.Context, query string, relay *bucket.Relay, options engi
 	pageRankCounter := make([]int, options.Pages.Max)
 
 	col.OnHTML(dompaths.Result, func(e *colly.HTMLElement) {
-		dom := e.DOM
+		linkText, titleText, descText := _sedefaults.FieldsFromDOM(e.DOM, dompaths, Info.Name)
 
-		linkHref, hrefExists := dom.Find(dompaths.Link).Attr("href")
-		linkText := parse.ParseURL(linkHref)
-		titleText := strings.TrimSpace(dom.Find(dompaths.Title).Text())
-		descText := strings.TrimSpace(dom.Find(dompaths.Description).Text())
-
-		if hrefExists && linkText != "" && linkText != "#" && titleText != "" {
-			pageIndex := _sedefaults.PageFromContext(e.Request.Ctx, Info.Name)
-			page := pageIndex + options.Pages.Start + 1
-
-			res := bucket.MakeSEResult(linkText, titleText, descText, Info.Name, page, pageRankCounter[pageIndex]+1)
-			bucket.AddSEResult(&res, Info.Name, relay, options, pagesCol)
+		pageIndex := _sedefaults.PageFromContext(e.Request.Ctx, Info.Name)
+		page := pageIndex + options.Pages.Start + 1
 
+		res := bucket.MakeSEResult(linkText, titleText, descText, Info.Name, page, pageRankCounter[pageIndex]+1)
+		valid := bucket.AddSEResult(&res, Info.Name, relay, &options, pagesCol)
+		if valid {
 			pageRankCounter[pageIndex]++
-		} else {
```

Comment:
Removed else logging on purpose?

Reply:
Well yeah it happens inside `bucket.AddSEResult` and `_sedefaults.FieldsFromDOM` now.